x-pack/plugins/alerting/server/rules_client/lib/siem_legacy_actions/migrate_legacy_actions.ts

```diff
@@ -103,11 +113,14 @@ export const migrateLegacyActions: MigrateLegacyActions = async (
     context.logger.error(
       `migrateLegacyActions(): Failed to migrate legacy actions for SIEM rule ${ruleId}: ${e.message}`
     );
-
-    return {
-      resultedActions: [],
-      hasLegacyActions: false,
-      resultedReferences: [],
-    };
+    throw Boom.badRequest(
```

Comment:
Just want to double-check that we're ok with the rulesClient function that is calling this bailing if this migration validation fails.

Reply:
The decision was to throw an exception and fail rule update, so that we don't have a data loss. Right now we just silently ignore the actions validation exception and return empty actions list in this case.